proto/streamlit/proto/BackMsg.proto

```diff
@@ -49,5 +49,8 @@ message BackMsg {
 
     // Set to true to ask the server to close the connection
     bool close_connection = 10;
+
+    // Request to pass current URL to backend
+    string url_info = 11;
```

Comment:
I have used the `store_url` for now (sorry for the weird name 😉 )

Rational: the only BackMsg being sent to Server is upon user first opens the tab. It would make sense the sent the browser URL back to Server. Server will do the parsing and store the query strings only (such we don't need to parse on frontend)